lib/rules/name-property-casing.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * @fileoverview Name property casing for consistency purposes
+ * @author Armano
+ */
+'use strict'
+
+const utils = require('../utils')
+
+function kebabCase (str) {
+  return str.replace(/([a-z])([A-Z])/g, match => match[0] + '-' + match[1]).replace(/\s+/g, '-').toLowerCase()
+}
+
+function camelCase (str) {
+  return str.replace(/(?:^\w|[A-Z]|\b\w)/g, (letter, index) => index === 0 ? letter.toLowerCase() : letter.toUpperCase()).replace(/[\s-]+/g, '')
+}
+
+function pascalCase (str) {
+  str = camelCase(str)
+  return str.length > 0 ? str.charAt(0).toUpperCase() + str.slice(1) : ''
+}
+
+function convertCase (str, caseType) {
+  if (caseType === 'kebab-case') {
+    return kebabCase(str)
+  } else if (caseType === 'PascalCase') {
+    return pascalCase(str)
+  }
+  return camelCase(str)
+}
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+function create (context) {
+  const options = context.options[0]
+  const caseType = ['camelCase', 'kebab-case', 'PascalCase'].indexOf(options) !== -1 ? options : 'PascalCase'
```

Comment:
Please move the array with possible caseTypes out of the `create` function, it's how it's done in most rules I think.